src/gateways/hosted_zone_gateway.py

```diff
@@ -0,0 +1,31 @@
+from os.path import commonprefix
+
+
+class HostedZoneGateway:
+    def __init__(self, client):
+        self.client = client
+
+    def get_hosted_zone(self, target_domain):
+        target_domain = self._sanitize_domain(target_domain)
+        zones = self.client.list_hosted_zones_by_name()
+        for zone in zones['HostedZones']:
+            if self._hosted_zone_matches_domain(zone, target_domain):
+                return zone
```

Comment:
We should define this return instead of sending the Boto object as-is.

Reply:
agreed. As it's a common problem across all gateways, we can spin this out in another PR